cppcache/src/Log.cpp

```diff
@@ -107,334 +78,280 @@ void Log::init(LogLevel level, const char* logFileName, int32_t logFileLimit,
+  // Must specify at least 1!
+  maxFiles = maxFiles > 0 ? maxFiles : 0;
+
+  return maxFiles;
+}
+
+spdlog::level::level_enum geodeLogLevelToSpdlogLevel(LogLevel logLevel) {
+  auto level = spdlog::level::level_enum::off;
+  switch (logLevel) {
+    case LogLevel::None:
+      level = spdlog::level::level_enum::off;
+      break;
+    case LogLevel::Error:
+      level = spdlog::level::level_enum::err;
+      break;
+    case LogLevel::Warning:
+      level = spdlog::level::level_enum::warn;
+      break;
+    case LogLevel::Info:
+      level = spdlog::level::level_enum::info;
+      break;
+    case LogLevel::Default:
+      level = spdlog::level::level_enum::info;
+      break;
+    case LogLevel::Config:
+      level = spdlog::level::level_enum::info;
+      break;
+    case LogLevel::Fine:
+      level = spdlog::level::level_enum::debug;
+      break;
+    case LogLevel::Finer:
+      level = spdlog::level::level_enum::debug;
+      break;
+    case LogLevel::Finest:
+      level = spdlog::level::level_enum::debug;
+      break;
+    case LogLevel::Debug:
+      level = spdlog::level::level_enum::debug;
+      break;
+    case LogLevel::All:
+      level = spdlog::level::level_enum::debug;
+      break;
   }
+
+  return level;
 }
 
-void Log::init(LogLevel level, const std::string& logFileName,
-               int32_t logFileLimit, int64_t logDiskSpaceLimit) {
-  if (g_log != nullptr) {
-    throw IllegalStateException(
-        "The Log has already been initialized. "
-        "Call Log::close() before calling Log::init again.");
+void Log::init(LogLevel logLevel, const std::string& logFilename,
+               uint32_t logFileSizeLimit, uint64_t logDiskSpaceLimit) {
+  if (logLevel == LogLevel::None) {
+    return;
   }
-  s_logLevel = level;
 
   try {
     std::lock_guard<decltype(g_logMutex)> guard(g_logMutex);
 
-    g_hostName = boost::asio::ip::host_name();
+    if (logFilename.empty()) {
+      if (logFileSizeLimit || logDiskSpaceLimit) {
+        IllegalArgumentException ex(
+            "Cannot specify a file or disk space size limit without specifying "
+            "a log file name.");
+        throw ex;
+      }
+      currentLevel = logLevel;
+      // Check for multiple initialization
+      if (currentLogger && currentLogger->name() == "console") {
+        return;
+      } else {
+        close();
+      }
+      currentLogger = spdlog::stderr_color_mt("console");
+      currentLogger->set_level(geodeLogLevelToSpdlogLevel(currentLevel));
+      currentLogger->set_pattern(logLineFormat());
+      return;
+    } else if (logDiskSpaceLimit && logFileSizeLimit > logDiskSpaceLimit) {
+      IllegalArgumentException ex(
+          "File size limit must be smaller than disk space limit for "
+          "logging.");
+      throw ex;
+    }
 
-    g_fullpath =
-        boost::filesystem::absolute(boost::filesystem::path(logFileName));
+    int32_t adjustedFileSizeLimit;
```

Comment:
spdlog has its own log rotation, why are we implementing our own?

Reply:
We're not... really.  Our two existing properties are file size and disk space, specified in MB.  spdlog uses two properties, file size and max number of files, specified in bytes.  So we calculate limits like we did before, then translate disk space to a max number of files and give those limits to the spdlog file sink.  There's no logic anymore in our code to check file size, roll logs, delete old files, etc.